File: Lab Vezba 1/zad 1/Automobil.java
import izuzeci.VoziloSeKreceIzuzetak;

public class Automobil extends Vozilo{
	boolean stanjeVrata;
	
	public Automobil(int maxBrzina, int trenutnaBrzina, int potrosnja) {
		stanjeVrata = false;
		stanje = false;
		this.maxBrzina = maxBrzina;
		this.trenutnaBrzina = trenutnaBrzina;
		this.potrosnja = potrosnja;
	}
	
	public void otvoriVrata() throws VoziloSeKreceIzuzetak{
		if(trenutnaBrzina == 0) {
			stanjeVrata = true;
		} else {
			throw new VoziloSeKreceIzuzetak("Nemoguce je otvoriti vrata dok je vozilo u pokretu!");
		}
		
	}
	
	public void zatvoriVrata() {
		stanjeVrata = false;
	}
	
	@Override
	public void ubrzaj() {
		trenutnaBrzina += 10;
		if(trenutnaBrzina > maxBrzina) {
			trenutnaBrzina = maxBrzina;
		}
	}

	@Override
	public void uspori() {
		trenutnaBrzina -= 30;
		if(trenutnaBrzina < 0) {
			trenutnaBrzina = 0;
		}
	}

}

File: LV-4/zad1/Automobil.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp1
{
    class Automobil : Vozilo
    {
        private StanjeVrata[] vrata;

        public Automobil(StanjeVozila stVoz, int maxBrzina, int trenutnaBrzina , double potrosnja)
        {
            this.stVoz = stVoz;
            this.maxBrzina = maxBrzina;
            this.trenutnaBrzina = trenutnaBrzina;
            this.potrosnja = potrosnja;
            vrata = new StanjeVrata[4];
        }

        public Automobil(Automobil a)
        {
            vrata = new StanjeVrata[4];
            for(int i = 0; i < 4; i++)
            {
                vrata[i] = a.StVrata[i];
            }
            this.stVoz = a.stVoz;
            this.maxBrzina = a.maxBrzina;
            this.trenutnaBrzina = a.trenutnaBrzina;
            this.potrosnja = a.potrosnja;
        }

        public override StanjeVrata[] StVrata {
            get { return vrata; }
            set { this.vrata = value; }
        }

        public void otvoriVrata(int i)
        {
            vrata[i] = StanjeVrata.otvorena;
        }

        public void zatvoriVrata(int kojaVrata)
        {
            vrata[kojaVrata] = StanjeVrata.zatvorena;
        }

        public static Automobil operator ++(Automobil a)
        {
            a.trenutnaBrzina += 10;
            return a;
        }

        public static Automobil operator --(Automobil a)
        {
            a.trenutnaBrzina -= 30;
            return a;
        }
    }
}
